src/main/java/com/ea/restaurant/service/impl/UserServiceImpl.java

```diff
@@ -0,0 +1,64 @@
+package com.ea.restaurant.service.impl;
+
+import com.ea.restaurant.constants.Status;
+import com.ea.restaurant.entities.User;
+import com.ea.restaurant.exceptions.EntityNotFoundException;
+import com.ea.restaurant.exceptions.UserNotFoundException;
+import com.ea.restaurant.repository.UserRepository;
+import com.ea.restaurant.service.UserService;
+import java.util.List;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+@Service
+public class UserServiceImpl implements UserService {
+
+  @Autowired private UserRepository userRepository;
+
+  @Override
+  public User save(User user) {
+    return userRepository.save(user);
+  }
+
+  @Override
+  public List<User> findAllEnabled() {
+    return userRepository.findAllByEntityStatus(Status.ACTIVE);
+  }
+
+  @Override
+  public User findEnabledById(Long id) {
+    return userRepository
+        .findByIdAndEntityStatus(id, Status.ACTIVE)
+        .orElseThrow(() -> new EntityNotFoundException(id));
+  }
+
+  @Override
+  public void deleteUserById(Long id) {
+    User userToDelete = this.findEnabledById(id);
+    userToDelete.setEntityStatus(Status.DELETED);
+    this.save(userToDelete);
+  }
+
+  @Override
+  public User updateById(Long id, User user) {
+    User userToUpdate = this.findEnabledById(id);
+    userToUpdate.setName(user.getName());
+    userToUpdate.setLastName(
+        user.getLastName() == null ? userToUpdate.getLastName() : user.getLastName());
+    userToUpdate.setUsername(
+        user.getUsername() == null ? userToUpdate.getUsername() : user.getUsername());
+    userToUpdate.setPassword(
+        user.getPassword() == null ? userToUpdate.getPassword() : user.getPassword());
+    userToUpdate.setRoles(user.getRoles() == null ? userToUpdate.getRoles() : user.getRoles());
+    userToUpdate.setType(user.getType() == null ? userToUpdate.getType() : user.getType());
+    this.save(userToUpdate);
+    return userToUpdate;
+  }
+
+  @Override
+  public User findEnabledByUsername(String username) {
+    return userRepository
+        .findByUsernameAndEntityStatus(username, Status.ACTIVE)
+        .orElseThrow(() -> new UserNotFoundException(username));
```

Comment:
Shouldn't the developer that is using the service decide what to do if the user is not found?

Reply:
Done.